drivers/entropy/entropy_stm32.c

```diff
@@ -357,6 +357,21 @@ static int entropy_stm32_rng_init(const struct device *dev)
 	__ASSERT_NO_MSG(dev_cfg != NULL);
 
 #if CONFIG_SOC_SERIES_STM32L4X
+
+#if !defined(CONFIG_CLOCK_STM32_PLL_SRC_MSI) && \
+	!defined(CONFIG_CLOCK_STM32_PLL_SRC_HSI) && \
+	!defined(CONFIG_CLOCK_STM32_PLL_SRC_HSE)
```

Comment:
This is equivalent to `!CONFIG_CLOCK_STM32_SYSCLK_SRC_PLL`, which is exclusive with CONFIG_CLOCK_STM32_SYSCLK_SRC_HSI and CONFIG_CLOCK_STM32_SYSCLK_SRC_MSI.
So you don't need these 2 levels of #if's

Reply:
Thanks for this input, it should simplify a lot.